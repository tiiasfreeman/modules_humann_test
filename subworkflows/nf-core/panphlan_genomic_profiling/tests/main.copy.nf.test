nextflow_workflow {

    name "Subworkflow PANPHLAN_GENOMIC_PROFILING"
    script "subworkflows/nf-core/panphlan_genomic_profiling/main.nf"
    config "./nextflow.config"
    workflow "PANPHLAN_GENOMIC_PROFILING"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/panphlan_genomic_profiling"
    tag "panphlan/downloadpangenome"
    tag "panphlan/panphlan_map"
    tag "panphlan/panphlan_profiling"

    test("Staphylococcus aureus - paired end - works") {
        when {
            workflow {
                """
                input[0] = "Staphylococcus_aureus"
                input[1] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                        [file("/workspace/modules_humann_test/modules/local/SRA_data/Staph_aureus_SRR29085994.fastq.gz")]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
