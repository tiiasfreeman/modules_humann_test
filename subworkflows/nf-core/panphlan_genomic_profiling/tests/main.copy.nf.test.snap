{
    "Eubacterium_rectale - paired end - works": {
        "content": [
            {
                "0": [
                    "Eubacterium_rectale_pangenome.tsv:md5,c386350407e2787f4c108b73edeb49da"
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_Eubacterium_rectale_1.tsv:md5,4f3269cba7741128e6b8749c2e80a5bd"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "Eubacterium_rectale_profile.tsv:md5,f6619751444821d2e9ee1f4dfd7ec301"
                    ]
                ],
                "3": [
                    "versions.yml:md5,91040bc18a58ef8d1f8deb19414b74cf",
                    "versions.yml:md5,9743fc172c58fa40d3c30950b7844912",
                    "versions.yml:md5,9743fc172c58fa40d3c30950b7844912"
                ],
                "mapping_directory": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_Eubacterium_rectale_1.tsv:md5,4f3269cba7741128e6b8749c2e80a5bd"
                    ]
                ],
                "pangenome": [
                    "Eubacterium_rectale_pangenome.tsv:md5,c386350407e2787f4c108b73edeb49da"
                ],
                "panphlan_profile": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "Eubacterium_rectale_profile.tsv:md5,f6619751444821d2e9ee1f4dfd7ec301"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,91040bc18a58ef8d1f8deb19414b74cf",
                    "versions.yml:md5,9743fc172c58fa40d3c30950b7844912",
                    "versions.yml:md5,9743fc172c58fa40d3c30950b7844912"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-18T18:23:24.388768054"
    }
}